src/script/lua_api/l_metadata.cpp

```diff
@@ -229,6 +229,42 @@ int MetaDataRef::l_get_keys(lua_State *L)
 	return 1;
 }
 
+// get_bool(self, name)
+int MetaDataRef::l_get_bool(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkAnyMetadata(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	IMetadata *meta = ref->getmeta(false);
+	if (meta == NULL) {
+		lua_pushboolean(L, false);
+		return 1;
+	}
+
+	std::string str_;
+	const std::string &str = meta->getString(name, &str_);
+	lua_pushboolean(L, is_yes(str));
```

Comment:
My only gripe is with the usage of `is_yes` here. This has the potential to hide all kinds of mistakes by effectively coercing everything (numbers, strings) to booleans in a weird way.

A "strongly typed" alternative would be to explicitly check against `"true"` and `"false"` strings. If it is neither, throw an error (which should be documented).

Reply:
Now returns an error